File: server/app/models/db/user/user_response.py
from pydantic import BaseModel
from typing import List
from app.models.db.user.user_course_progress_response import UserCourseProgressResponse
from app.db.enums import AvailableCourse, Gender


class UserResponse(BaseModel):
    id: int
    email: str
    username: str
    first_name: str
    last_name: str | None = None
    gender: Gender
    current_course: AvailableCourse | None = None
    languages_learning: List[str]
    languages_learned: List[str]
    course_progresses: List[UserCourseProgressResponse]

File: client/types/response_models/UserResponse.ts
import { Gender, AvailableCourse, AvailableDialect } from "../enums";
import { UserCourseProgressResponse } from "./UserCourseProgressResponse";

export interface UserResponse {
    id: number;
    email: string;
    username: string;
    first_name: string;
    last_name: string | null;
    gender: Gender;
    current_course: AvailableCourse | null;
    current_dialect: AvailableDialect | null;
    languages_learning: string[];
    languages_learned: string[];
    courses_completed: string[];
    course_progresses: UserCourseProgressResponse[];
}
